### PR TITLE
PWN-4862 - The OTP code input screen does not meet the requirements

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4862

## Description of Work

PWN-4862 - fix input OTP screen
PWN-4993 - fix auto send of request

### CODE REVIEW

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/interactor/CreateWalletInteractor.kt [hunk 1]
```diff
@@ -8,10 +8,16 @@ import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
 class CreateWalletInteractor(
     private val gatewayServiceRepository: GatewayServiceRepository,
     private val signUpFlowDataRepository: SignUpFlowDataLocalRepository,
-    private val tokenKeyProvider: TokenKeyProvider,
+    private val tokenKeyProvider: TokenKeyProvider
 ) {
     class CreateWalletFailure(override val message: String) : Throwable(message)
 
+    val timer
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/CreateWalletInteractor.kt:15]: add explicit type here please

File: app/src/main/java/org/p2p/wallet/auth/ui/smsinput/NewSmsInputPresenter.kt [hunk 2]
```diff
@@ -45,12 +38,23 @@ class NewSmsInputPresenter(
             is OnboardingFlow.RestoreWallet -> restoreWalletInteractor.getUserPhoneNumber()
         }
         userPhoneNumber?.let { view.initView(it) }
+        connectToTimer()
+    }
+
+    private fun connectToTimer() {
+        createWalletInteractor.timer?.let { timer ->
+            launch {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/smsinput/NewSmsInputPresenter.kt:46]: no need for this

File: app/src/main/java/org/p2p/wallet/auth/ui/smsinput/NewSmsInputPresenter.kt [hunk 3]
```diff
@@ -136,49 +140,45 @@ class NewSmsInputPresenter(
...
-                        val userPhoneNumber = createWalletInteractor.getUserPhoneNumber()
-                            ?: throw IllegalStateException("User phone number cannot be null")
-                        createWalletInteractor.startCreatingWallet(
-                            userPhoneNumber = userPhoneNumber,
-                            isResend = true
-                        )
+        if (createWalletInteractor.resetCount >= MAX_RESENT_CLICK_TRIES_COUNT) {
+            view?.navigateToSmsInputBlocked(GeneralErrorTimerScreenError.BLOCK_SMS_RETRY_BUTTON_TRIES_EXCEEDED)
+        } else {
+            launch {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/smsinput/NewSmsInputPresenter.kt:146]: better to extract all of this to a method
indent is too big

File: ui-kit/src/main/res/color/button_lime_to_mountain_text.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (gslevinkov) [inline ui-kit/src/main/res/color/button_lime_to_mountain_text.xml:2]: i would rather to name this file with prefix `selector`
it's not a class to call it as button

File: app/src/main/java/org/p2p/wallet/auth/repository/SignUpFlowDataLocalRepository.kt [hunk 5]
```diff
@@ -16,10 +25,16 @@ private val TAG = SignUpFlowDataLocalRepository::class.simpleName.orEmpty()
  * and generated during the whole Create wallet onboarding flow
  */
 class SignUpFlowDataLocalRepository(
-    private val signUpUserStorage: UserSignUpDetailsStorage
+    private val signUpUserStorage: UserSignUpDetailsStorage,
+    private val appScope: AppScope
 ) {
 
+    private var smsResendTimers = listOf(30, 40, 60, 90, 120)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/SignUpFlowDataLocalRepository.kt:32]: it's not exactly timers, but `smsTimerStartSeconds`

File: app/src/main/java/org/p2p/wallet/auth/repository/SignUpFlowDataLocalRepository.kt [hunk 6]
```diff
@@ -16,10 +25,16 @@ private val TAG = SignUpFlowDataLocalRepository::class.simpleName.orEmpty()
...
+
     var userPhoneNumberEnteredCount = 0
+    var smsResendCount = 0
+
+    var timerFlow: Flow<Int>? = null
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/SignUpFlowDataLocalRepository.kt:37]: smsInputTimerFlow

File: app/src/main/java/org/p2p/wallet/auth/repository/SignUpFlowDataLocalRepository.kt [hunk 7]
```diff
@@ -54,6 +69,28 @@ class SignUpFlowDataLocalRepository(
             Timber.tag(TAG).i("User phone is received and set: ${userPhoneNumber?.formattedValue?.length}")
         }
 
+    fun startTimer() {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/repository/SignUpFlowDataLocalRepository.kt:72]: startSmsInputTimerFlow
it's not some abstract timer

**Review by gslevinkov — APPROVED** [review]

**Comment by gslevinkov** [comment]
And looks like all this logic with timer is used only on sms input fragment.
i suggest to put all of it to another pure fabrication entity, it would be wiser.

**Pull request merged**